In [1]:
import pandas as pd
import numpy as np

In [2]:
corpus = pd.read_csv("../../data/eval_corpus.csv", sep=";")
del corpus['Unnamed: 0']

In [3]:
set(corpus["query"])

{'a_hoax',
 'a_video',
 'aaron_is_dead',
 'aaron_swartz',
 'amber_alert',
 'american_airlines',
 'angelina_jolie',
 'armed_guards',
 'big_board',
 'blue_jackets',
 'cam_newton',
 "cartridges_didn't",
 'ces_2013',
 'chase_jarvis',
 'chip_kelly',
 'cirque_du_soleil',
 'cnn_chile',
 'colonial_marines',
 'david_miscavige',
 'debt_ceiling',
 'devil_may_cry',
 'duchess_of_cambridge',
 'el_riesgo_de',
 'fc_barcelona',
 'former_bodyguard',
 'fue_el',
 'golden_globes',
 'gun_control',
 'gun_violence',
 'i_know',
 'ice_cream',
 'indiana_jones',
 'internet_genius',
 'iphone_5s',
 'issued_for',
 "jack_lew's_signature",
 'jodie_foster',
 'justin_timberlake',
 'katherine_webb',
 'kris_humphries',
 'la_mega',
 'lance_armstrong',
 'lena_dunham',
 "lew's_signature_is",
 'met_office',
 'mike_mccoy',
 'miss_america',
 'new_maps',
 'nfl_kicks',
 'para_la',
 'penny_arcade',
 'return_to_music',
 'roger_ebert',
 'ryan_seacrest',
 'school_shooting',
 'serious_jibber-jabber',
 'set_for',
 'sky_sports',
 'sprin

In [4]:
# query: whats the point
corpus[corpus["query"] == "what's_the_point"]

source             query  score  \
550  reddit  what's_the_point      5   
551    news  what's_the_point      1   
552    news  what's_the_point      1   
553    news  what's_the_point      1   
554  reddit  what's_the_point     14   
555    news  what's_the_point      1   
556  reddit  what's_the_point     54   
557    news  what's_the_point      1   
558  reddit  what's_the_point     22   
559    news  what's_the_point      1   

                                              headline  \
550  If you're an atheist, what's the point in livi...   
551  <strong>FoxBusiness.com</strong> - Time to Pay...   
552                      The Oxford Eagle - Jimmy Reed   
553  <strong>Berkshire Eagle</strong> - Man charged...   
554           At what point does it stop being coffee?   
555  <strong>LA Times</strong> - December home pric...   
556  At what point do I disclose this somewhat cont...   
557  How To - Hit a Scissor Kick Overhead - <strong...   
558  Follow Up To: At what point do you just Give U...   
559  Another Day, Another Goldman Sachs Exec, Anoth...   

                                               summary  
550  Before the flaming starts, read what I have to...  
551  You&#39;ll want to get to <strong>the <strong>...  
552  Local columnist Jimmy Reed writes about a youn...  
553  A Guatemalan man living in Pittsfield is being...  
554  A lot of people say they like coffee, but what...  
555  The median is <strong>the <strong>point</stron...  
556  I know this belongs in /r/askmen but this sub ...  
557  When you&#39;re at net and a lob goes over you...  
558  Four days ago I posted [At what point do you j...  
559  (On this count, <strong>the <strong>point</str...

Score: -0.014798<br>
Score: -0.015701<br>
Score: -0.016491<br>
Score: -0.016642<br>
Score: -0.017701<br>
Score: -0.018240<br>
Score: -0.018658<br>
Score: -0.018713<br>
Score: -0.018824<br>
Score: -0.018914<br>
<br>
557 How To - Hit a Scissor Kick Overhead - tennis.com<br>
465 Serious question: When" can people "pirate" airline tickets?"<br>
554 At what point does it stop being coffee?<br>
558 Follow Up To: At what point do you just Give Up"? "<br>
24 Messi renews Barcelona contract - Taipei Times<br>
101 What's the weather like where you are?<br>
819 Is ironman mode a new concept in gaming that I've missed until now?<br>
129 Philly.com - The Numbers Behind Billick - BoopStats<br>
339 Pats hope to continue success without Gronkowski - NewsOK.com<br>
290 Super Bowl Predictions: Must include justification.<br>
<br>
=>20% accuracy<br>



In [11]:
# Query: aaron_is_dead
corpus[corpus["query"] == "aaron_is_dead"]

source          query  score  \
590  reddit  aaron_is_dead      1   
591  reddit  aaron_is_dead      7   
592  reddit  aaron_is_dead     10   
593  reddit  aaron_is_dead      5   
594  reddit  aaron_is_dead     96   
595  reddit  aaron_is_dead     12   
596  reddit  aaron_is_dead      9   
597  reddit  aaron_is_dead      5   
598  reddit  aaron_is_dead     10   
599  reddit  aaron_is_dead      9   

                                              headline  \
590  Who should I pick up from the waiver wire: Cra...   
591                               Laying a Trap Part 2   
592  Favorite Fight Friday: Aaron Pryor vs Alexis A...   
593  [KEEPER ADVICE] - At what point do you throw i...   
594         Aaron Shwartz is dead - Reddit, we did it!   
595    So is Bruce Banner Back? (Incredible Hulk #7.1)   
596  [Story 1] The Greater Thelema Society - A Disp...   
597       Aaron's Android Advice reviews: Mage Defense   
598  Name 3 people dead or alive that you would lov...   
599  [Contains Huge Comic Spoilers]: Fan speculatio...   

                                               summary  
590  I plan on dropping Victor Cruz for one of thes...  
591  2nd Squadron surged from the Thermopylae, secu...  
592  With all the resurgence of the Pacweather talk...  
593  Hey all. In my PPR league, I'm now 3-6 and dea...  
594                                                  .  
595  So 7 issues in and Bruce Banner is already bac...  
596  “It’s about time you showed up.”\n\n“Don’t act...  
597  Aaron’s Android Advice\n\nTITLE:\tMage Defense...  
598  My room mate and I got vaped tonight and decid...  
599  So I was having a discussion about TWD with th...

Score: -0.020255<br>
Score: -0.034791<br>
Score: -0.034818<br>
Score: -0.036768<br>
Score: -0.037668<br>
Score: -0.038481<br>
Score: -0.039062<br>
Score: -0.039318<br>
Score: -0.039349<br>
Score: -0.039872<br>
<br>
594 Aaron Shwartz is dead - Reddit, we did it!<br>
305 How do you rank the 32 starting Quarterbacks?<br>
506 RIP Aaron Swartz<br>
357 New Game Releases – January 2013<br>
85 PETER DINKLAGE WINS THE GOLDEN GLOBE!!!<br>
350 New Video Game Releases – January 2013<br>
300 Over 1/3 of starting QB's are in their rookie or sophomore season.<br>
610 [H] Skyrim, Arma 2 CO, CS:GO, Walking Dead, DmC Devil May Cry, Sid Meier's Civilization V - Gods and Kings, Arma X, FM 2013, Dishonored, Blackops 2 [W] TF Keys, Offers <br>
368 Miss Alabama Katherine Webb thinks being called 'Sexy' is derogatory.<br>
356 New Game Releases – January 2013 - Q 103 - Albany's #1 Rock Station<br>
=>20% Accuracy

In [12]:
# justin_timberlake
corpus[corpus["query"] == "justin_timberlake"]

source              query  score  \
690    news  justin_timberlake      1   
691    news  justin_timberlake      1   
692    news  justin_timberlake      1   
693    news  justin_timberlake      1   
694  reddit  justin_timberlake     19   
695    news  justin_timberlake      1   
696    news  justin_timberlake      1   
697    news  justin_timberlake      1   
698    news  justin_timberlake      1   
699    news  justin_timberlake      1   

                                              headline  \
690  <strong><strong>Justin</strong> <strong>Timber...   
691     <strong>99.5 Magic FM</strong> - From ABC News   
692  <strong>Sun Herald</strong> - Highlights of Ho...   
693  Tip(s) of the hat - <strong>StarTribune.com</s...   
694  How come musicians often become well-respected...   
695  <strong>Front</strong> - Highlights of Hollywo...   
696  <strong>Ohio</strong> - Oscar Documentary Semi...   
697  Product News - Panasonic Televisions to Offer ...   
698  Forget about the Oscars, <strong><strong>Justi...   
699  <strong>WKBT</strong> - Celebrities with Janua...   

                                               summary  
690  Get The Commercial Appeal on your iPhone, iPad...  
691  There is a precedent for musicians who&#39;ve ...  
692  &quot;Runner Runner&quot;: An online gambling ...  
693  <strong><strong>Justin</strong> <strong>Timber...  
694  Mark Wahlberg, Justin Timberlake, Will Smith, ...  
695  &quot;Runner Runner&quot;: An online gambling ...  
696  List is here I expect to have more to say late...  
697  Music Star <strong><strong>Justin</strong> <st...  
698  ONLINE TRENDS FOR JAN 10, 2013.  By Tim Swift,...  
699  If January has you celebrating a birthday, you...

Score: -0.045070<br>
Score: -0.049264<br>
Score: -0.055307<br>
Score: -0.057209<br>
Score: -0.057606<br>
Score: -0.058397<br>
Score: -0.060346<br>
Score: -0.060892<br>
Score: -0.062312<br>
Score: -0.062392<br>
<br>
321 Friday's TV talk shows - StarTribune.com<br>
594 Aaron Shwartz is dead - Reddit, we did it!<br>
641 NEW YORK - Timberlake hints return to music in video - The Sun Herald - Celebrities<br>
305 How do you rank the 32 starting Quarterbacks?<br>
325 KTVO-TV 3 - Justin Bieber gets love at American Music Awards - The latest local news, weather and high ...<br> 
320 HeraldNet.com - Premieres, finales abound on January TV schedule - HeraldNet.com - Entertainment<br>
762 Belfast, Northern Ireland, UK, World, News, Business, Entertainment, belfasttelegraph.co.uk - Independent Woman - Belfasttelegraph.co.uk<br>
264 Lance Armstrong Interview Could Be Worth Millions to Oprah Winfrey Network - Forbes<br>
691 99.5 Magic FM - From ABC News<br>
379 WFMY News 2 - Naomi Campbell Mugged, George Lucas Engaged - Video - digtriad.com<br>
=>Accuracy 10%

In [14]:
# devil_may_cry
corpus[corpus["query"] == "devil_may_cry"]

source          query  score  \
350    news  devil_may_cry      1   
351    news  devil_may_cry      1   
352    news  devil_may_cry      1   
353  reddit  devil_may_cry    960   
354    news  devil_may_cry      1   
355  reddit  devil_may_cry      8   
356    news  devil_may_cry      1   
357    news  devil_may_cry      1   
358  reddit  devil_may_cry      5   
359  reddit  devil_may_cry     27   
610  reddit  devil_may_cry      2   
611    news  devil_may_cry      1   
612  reddit  devil_may_cry    584   
613    news  devil_may_cry      1   
614  reddit  devil_may_cry     15   
615  reddit  devil_may_cry      4   
616  reddit  devil_may_cry     29   
617    news  devil_may_cry      1   
618    news  devil_may_cry      1   
619  reddit  devil_may_cry      3   

                                              headline  \
350             New Video Game Releases – January 2013   
351  DMC - <strong><strong>Devil</strong> <strong><...   
352                       Too Human Review - MLive.com   
353  Steam Summer Camp Sale Day 2: Total War Shogun...   
354  <strong>Chicago Tribune</strong> - Monday melt...   
355            WSIB: If I Liked the Devil May Cry Demo   
356  New Game Releases – January 2013 - Q 103 - Alb...   
357                   New Game Releases – January 2013   
358  [USA] [H] XCOM: Enemy Unknown, Dishonored, Bor...   
359  IAmA music producer for Sony Music Japan and h...   
610  [H] Skyrim, Arma 2 CO, CS:GO, Walking Dead, Dm...   
611  <strong>LEO Weekly</strong> - Video TapeWorm -...   
612               This Week in Gaming: October 16 - 22   
613  <strong>ContraCostaTimes.com</strong> - A look...   
614  /r/Gaming, what video game do you think revolu...   
615  [USA] [H] Kinect, list (360), Dark Souls, God ...   
616  IAmA music producer for Sony Music Japan and h...   
617  New screenshots released for <strong><strong>D...   
618  <strong>Austin's source for local news</strong...   
619  [USA] [H] Borderlands 2, Darksiders 2, Dead or...   

                                               summary  
350  <strong><strong>Devil</strong> <strong><strong...  
351  <strong><strong>Devil</strong> <strong><strong...  
352  Sure Too Human could have been a button masher...  
353  Me again.  \n  \nhttp://store.steampowered.com...  
354  Taking a cue from &quot;Silent Hill,&quot; the...  
355    I just installed the Devil May Cry Demo and ...  
356  <strong><strong>Devil</strong> <strong><strong...  
357  <strong><strong>Devil</strong> <strong><strong...  
358  Games are in **pristine** condition, and come ...  
359  This AMA is for my gf. She's worked at Sony Mu...  
610  * [ARMA II: Combined Operations](http://store....  
611  The aftermath <strong>may</strong> be the most...  
612  [**Mobile Version**](https://sites.google.com/...  
613  The fun starts immediately in January with the...  
614  Just curious what everyone's opinions are on t...  
615  **Haves**\n\n*Xbox 360:* \n\n* ~~Kinect device...  
616  This AMA is for my gf. She's worked at Sony Mu...  
617  The Kids are Alright in Dungeon Defenders.  Sl...  
618  I&#39;m wondering if it&#39;s going to be kind...  
619  **Games I have**\n\n\n360:\n\n\n* borderlands ...

Score: 0.000917<br>
Score: -0.001187<br>
Score: -0.003866<br>
Score: -0.004415<br>
Score: -0.004555<br>
Score: -0.005259<br>
Score: -0.005286<br>
Score: -0.008320<br>
Score: -0.009146<br>
Score: -0.010330<br>

352 Too Human Review - MLive.com<br>
351 DMC - Devil May Cry's Sexy Character Reveal — Kat<br>
357 New Game Releases – January 2013<br>
76 Can we call progamers" something else?<br>
618 Austin's source for local news - Joystick takes an early look at Deadpool game - The Austin Daily Herald<br>
350 New Video Game Releases – January 2013<br>
355 WSIB: If I Liked the Devil May Cry Demo<br>
610 [H] Skyrim, Arma 2 CO, CS:GO, Walking Dead, DmC Devil May Cry, Sid Meier's Civilization V - Gods and Kings, Arma X, FM 2013, Dishonored, Blackops 2 [W] TF Keys, Offers <br>
356 New Game Releases – January 2013 - Q 103 - Albany's #1 Rock Station<br>
775 PCMag.com - Web Mourns Internet Activist Aaron Swartz, 26 - News &amp; Opinion - PCMag.com<br>
=> Accuracy 80%